### ISSUE TITLE
Negative spacing

### ISSUE THREAD

**Issue by fred-wang** [mathml-refresh/mathml#40]
"The present document assumes that W,w,a,b,A,BW,w,a,b,A,B, italic correction and top accent attachment are nonnegative values. If computations result in negative values, then they must be interpreted as being zero. Note that some MathML elements such as mspace or mpadded may set box dimensions to negative values."
In the future, we should explain how to handle negative spacing.

original report: https://gitlab.com/mathml/MathMLinHTML5/issues/31

**Comment by NSoiffer** [comment]
Negative horizontal values can be tricky around linebreaks as they may not be intended if not paired with their neighbors. Other than that, what problems do you see with negative values? My memory from my MathPlayer implementation days is that negative values do occur for some of those values and are reasonable.

It would be good to enumerate in this issue the problems that happen if a value is negative rather than just force them to be zero. Forcing them to zero is probably not right for at least some of these values. For example, "-" has a negative _depth_ ("b") and it is likely wrong to make the element taller by forcing the depth to be 0.

**Comment by fred-wang** [comment]
I agree negative spacing is needed. The basic implementation of `mspace` is just setting a box of the desired dimensions. I'm not sure you can specify negative box sizes in layout so this would need to be handled differently (probably implemented as negative margin?). Again, I believe this is something to carefully analyzed with web engines' implementers.

**Comment by physikerwelt** [comment]
I hope that there is a solution that we can simply adopt. Maybe SVG has already a method for negative spaces.

**Comment by fred-wang** [comment]
Might related to #14

**Comment by fred-wang** [comment]
With the current writing of the spec, the lspace/rspace are no longer part of the content box of the embellished op but are added in mrow so negative spacing is easier to describe. Maybe we should have the same lspace/rspace box properties for mspace or mpadded and generalize to any box.
